main.cpp

```diff
@@ -125,419 +152,429 @@ struct CarWash
+    if( engineHorsePower > 2000)
+    {
+        isFlying = true;
+    }
+    return isFlying;
+}
+
+
+
 struct Oscillator
 {
-    //Tuning (semitones)
-    int semitones;
-    //Waveform
-    char Waveform;
-    //Phase
+    int semitone;
+    char waveform;
     int phaseDegree;
-    // Noise level
     float noiseLevel;
-    //LFO amount
     int lfoAmount;
 
-    //generate signal
     void outputSignal(char channel, int amplitude);
-    //Sync with another oscillator
     void syncOscillator(Oscillator oscToSyncTo);
-    //Modulate other parameter
     void outputAsLFO(std::string targetParameter);
 };
-/*
-Thing 6) Filter
-5 properties:
-    1) Cutoff
-    2) Resonance
-    3) Type
-    4) Envelope amount
-    5) Wet/dry
-3 things it can do:
-    1)  Switch state
-    2)  Switch inputs
-    3)  Sweep up/down
- */
+
+void Oscillator::outputSignal(char channel, int amplitude)
+{
+    for(int i = 0; i < 2; i++)
+    {
+        channel = 'L';
+        amplitude = 75 * i;
+    }
+}
+void Oscillator::syncOscillator(Oscillator oscToSyncTo)
+{
+    phaseDegree = oscToSyncTo.phaseDegree;
+}
+void Oscillator::outputAsLFO(std::string targetParameter)
+{
+    targetParameter = "LFO output target";
+}
+
+
 struct Filter
 {
-    //Cutoff freq
     int cutoffFreq;
-    // Resonance
     int resonance;
-    // Type
     char filterType;
-    // Envelope amount
     int envAmount;
-    // Wet/dry
     int wet;
 
     struct LFO
     {
-        //LFO frequency
         float frequency;
-        // bpm ratio of the LFO frequency
         int tempoMultiplier;
-        //LFO waveform
         int waveform;
-        //Makes waveform sharper/smoother
         float waveformMorph;
-        //tells whether lfo is synced to bpm
         bool bpmSync;
 
-        //change waveform
         void changeWaveform(int nextWave);
-        //change LFO bpm ratio
         void changeTempoMultiplier(int newValue);
-        //sync LFO to Oscillator
         void syncToOscillator(Oscillator syncTo, float warpAmount);
     };
 
-    // Switch state
     void changeType(char nextType);
-    //Switch inputs
     void switchInput(int newInput);
-    //Sweep up/down
-    void filterSweep(int startFreq, int endFreq);
+    void filterSweep(int startFreq, int endFreq, float sweepTimeInMillis);
 };
-/*
-Thing 7) Amplifier
-5 properties:
-    1) Drive level 
-    2) Tone
-    3) Waveshaper type
-    4) bass boost (int)
-    5) Output level 
-3 things it can do:
-    1) Attenuate signal
-    2) Saturation
-    3) Change waveshaper
- */
+
+void Filter::changeType(char nextType)
+{
+    filterType = nextType;
+}
+void Filter::switchInput(int newInput)
+{
+    newInput += 1;
+    if(newInput > 5) newInput = 0;
+}
+void Filter::filterSweep(int startFreq, int endFreq, float sweepTimeInMillis)
+{
+    for(float f = 0.0f; f < sweepTimeInMillis; f += 0.5f)
+    {
+        cutoffFreq = startFreq;
+        if(cutoffFreq < endFreq) cutoffFreq++;
+
+        else  cutoffFreq--;
```

Comment:
always prefer Pre-increment instead of Post-increment. 
if you need an explanation, send me a message in slack.
Also, it's generally more readable if the `if()` body is on its own line when its a single-line expression:
```
if( expr )
    singleLineExpression
else
    singleLineExpression
```
it's just easier to *see* the logic.
- [ ] revise